sycl/doc/extensions/SPIRV/SPV_INTEL_arbitrary_precision_fixed_point.asciidoc

```diff
@@ -0,0 +1,470 @@
+
+After Section 3.16, add a new section "3.16b Quantization Modes" as follows:
+
+=== Quantization Modes
+[options="header"]
+[width="80%"]
+[cols="15%,20%,65%"]
+|====
+| Value ^| Mode        ^| Behavior
+| 0      | TRN          | Truncate towards -Inf
+| 1      | TRN_ZERO     | Truncate towards 0
+| 2      | RND          | Round towards +Inf
+| 3      | RND_ZERO     | Round towards 0
+| 4      | RND_INF      | Round towards +/-Inf
+| 5      | RND_MIN_INF  | Round towards -Inf
+| 6      | RND_CONV     | Round towards even
+| 7      | RND_CONV_ODD | Round towards odd
+|====
+
+After Section 3.16, add a new section "3.16c Overflow Modes" as follows:
+
+=== Overflow Modes
+[options="header"]
+[width="80%"]
+[cols="15%,20%,65%"]
+|====
+| Value ^| Mode     ^| Behavior
+| 0      | WRAP      | Drop the bits to the left of the MSB
+| 1      | SAT       | Saturate to the closest of MIN or MAX
+| 2      | SAT_ZERO  | Set to 0 on overflow
+| 3      | SAT_SYM   | For unsigned, treat as SAT. +
+
+For signed: a positive overflow will saturate at the maximum positive value, whereas a negative overflow will saturate to the negation of the maximum positive value, as opposed to the most negative value.
+|====
+
+=== Capability
+Modify Section 3.31, *Capability*, adding a row to the Capability table:
+
+[options="header"]
+|====
+2+^| Capability ^| Implicitly Declares
+| 5922 | *ArbitraryPrecisionFixedPointINTEL* +
+
+Enables arbitrary precision fixed-point math instructions. +
+
+Note that this fixed-point type does not hold information about the sign in the type, but the information is rather encoded into the operations performed on this type.
+| *INTELArbitraryPrecisionIntegers*, *Kernel*
+|====
+
+=== Instructions
+
+In Section 3.32.13, *Arithmetic Instructions*, add the following instructions:
+
+[cols="10", width="100%"]
+|=====
+9+<|*OpFixedSqrtINTEL* +
+
+An `OpTypeInt` representing an arbitrary precision fixed point number (ac_fixed) is passed in as the _Input_ and the square root of the value is returned in _Result_.
+The behavior of this function is undefined for input values < 0.
+
+_Result Type_ is an `OpTypeInt` of width _rW_ and is the type of _Result_.
+
+_S_ is an integer literal, treated as a boolean, that indicates if the input and output are signed or unsigned.
+A value of `1` indicates that the two types are signed.
+
+_I_ is an integer literal that refers to the location of the fixed-point relative to the MSB of the input.
+
+_rI_ is an integer literal that refers to the location of the fixed-point relative to the MSB of the result.
+
+_Q_ is an integer literal chosen from _Table 3.16b_ that indicates the Quantization Mode of this operation.
+
+_O_ is an integer literal chosen from _Table 3.16c_ that indicates the Overflow Mode of this operation.
+
+| Capability:
+*ArbitraryPrecisionFixedPointINTEL*
+
+| 9 | 5923 | <id> Result Type | Result <id> | Input <id> | _Literal_ S | _Literal_ I | _Literal_ rI | _Literal_ Q | _Literal_ O
```

Comment:
`Q` and `O` shan't be `Literal`, instead they shall be `QuantizationModes` and `OverflowModes` appropriately. It will enforce implementation to add these two enums to spirv.hpp. As an example you can see `FPRoundingMode` decoration, that uses not `Literal` for its parameter, but value of `FP Rounding Mode` enum.

Reply:
@MrSidims does this look like as follows: 

```
_Q_ is a QuantizationMode enum chosen from _Table 3.16b_ that indicates the Quantization Mode of this operation.

_O_ is an OverflowMode enum chosen from _Table 3.16c_ that indicates the Overflow Mode of this operation.
...
| 9 | 5923 | <id> Result Type | Result <id> | Input <id> | _Literal_ S | _Literal_ I | _Literal_ rI | _QuatntizationMode_ Q | _OverflowMode_ O
```

Reply:
Yeap, sorry for a long response